bin/strata-cli/src/signet/persist.rs

```diff
@@ -0,0 +1,63 @@
+use std::{cell::RefCell, path::PathBuf, rc::Rc, sync::OnceLock};
+
+use bdk_wallet::{
+    rusqlite::{self, Connection},
+    ChangeSet, WalletPersister,
+};
+
+use crate::signet::SignetWallet;
+
+/// Wrapper around the built-in rusqlite db that allows
+/// [`PersistedWallet`](crate::signet::PersistedWallet) to be shared across multiple threads by
+/// lazily initializing per core connections to the sqlite db and keeping them in local thread
+/// storage instead of sharing the connection across cores.
+///
+/// WARNING: [`set_data_dir`] **MUST** be called and set before using [`Persister`].
+#[derive(Debug)]
+pub struct Persister;
+
+static DATA_DIR: OnceLock<PathBuf> = OnceLock::new();
+
+/// Sets the data directory static for the thread local DB.
+///
+/// Must be called before accessing [`Persister`].
+///
+/// Can only be set once - will return whether value was set.
+pub fn set_data_dir(data_dir: PathBuf) -> bool {
+    DATA_DIR.set(data_dir).is_ok()
+}
+
+thread_local! {
+    static DB: Rc<RefCell<Connection>> = RefCell::new(Connection::open(SignetWallet::db_path("default", DATA_DIR.get().expect("data dir to be set"))).unwrap()).into();
+}
```

Comment:
This data should be passed around as struct fields along with the data associated with it, not in globals/tls.